### PR TITLE
refactor(ChangeStream): use maybePromise for next/hasNext

### PR DESCRIPTION
**Pull request by emadum**
## Description

[NODE-2570](https://jira.mongodb.org/browse/NODE-2570)

**What changed?**

**Are there any files to ignore?**

### CODE REVIEW

File: test/functional/change_stream.test.js [hunk 1]
```diff
@@ -1933,7 +1933,7 @@ describe('Change Streams', function() {
     }
   });
 
-  describe('should properly handle a changeStream event being processed mid-close', function() {
+  describe.skip('should properly handle a changeStream event being processed mid-close', function() {
```
Comment (emadum) [inline test/functional/change_stream.test.js:1936]: These tests fail now, which is a little concerning, but I'm not 100% they are written correctly. They were already flaky, e.g. [see here](https://evergreen.mongodb.com/task/mongo_node_driver_stable_rhel70_boron_test_4.0_replica_set_unified_8a69ed696f4bf8d8790874118b72518d366b8fad_20_04_23_18_46_05).

File: test/functional/change_stream.test.js [hunk 2]
```diff
@@ -1982,7 +1982,7 @@ describe('Change Streams', function() {
       }
     });
 
-    it('when invoked with callbacks', {
+    it.skip('when invoked with callbacks', {
```
Comment (emadum) [inline test/functional/change_stream.test.js:1985]: These two tests fail; the callbacks-based one consistently, and the promises-based on intermittently, which is a little concerning, but I'm not 100% they are written correctly.

They were already flaky, e.g. [see here](https://evergreen.mongodb.com/task/mongo_node_driver_stable_rhel70_boron_test_4.0_replica_set_unified_8a69ed696f4bf8d8790874118b72518d366b8fad_20_04_23_18_46_05).

I'd propose fixing them (if desired) be moved into a separate ticket.

**Comment by ZYCGary** [comment]
The circular dependency warning still occur in my web app, just wanna know when could it be fixed?

File: test/functional/change_stream.test.js (hunk 2 above)
Comment (mbroadst) [reply test/functional/change_stream.test.js:1985]: SGTM

**Comment by mbroadst** [comment]
@ZYCGary the issue has already been reported and fixed, tracked by this JIRA ticket:  [NODE-2536](https://jira.mongodb.org/browse/NODE-2536). Please make a JIRA ticket instead of leaving comments on random pull requests.

**Pull request merged**